File: 1385. Find the Distance Value Between Two Arrays.java
// Easy
// Array, Simulation
// Time: O(mn), Space: O(1) 
// https://leetcode.cn/problems/find-the-distance-value-between-two-arrays/

class Solution {
  public int findTheDistanceValue(int[] arr1, int[] arr2, int d) {
      int count = arr1.length;
      for (int a : arr1) {
          for (int b : arr2) {
              if (Math.abs(a - b) <= d) {
                  count--;
                  break;
              }
          }
      }
      return count;
  }
}

File: 1001-2000/1385. Find the Distance Value Between Two Arrays.py
# Time:O(nm), Space:O(1)

from typing import List

class Solution:
    def findTheDistanceValue(self, arr1: List[int], arr2: List[int], d: int) -> int:
        ans = 0
        for x in arr1:
            if all(abs(x - y) > d for y in arr2):
                ans += 1
                
        return ans

# Can optimize use binary search
